src/v/security/audit/audit_log_manager.cc

```diff
@@ -390,18 +406,37 @@ ss::future<> audit_client::create_internal_topic() {
     }
 }
 
-ss::future<> audit_client::shutdown() {
-    /// Repeated calls to shutdown() are possible, although unlikely
+void audit_client::shutdown() {
     if (_as.abort_requested()) {
-        co_return;
+        return;
     }
     vlog(adtlog.info, "Shutting down audit client");
+    ss::abort_source exit;
+    auto abort = ss::sleep_abortable(3s, exit)
```

Comment:
maybe I'm missing it, but if the queue drain interval is >3 seconds, is it possible that this timeout will complete before a drain interval is kicked off?  Does something force the drain interval to kick off?

Reply:
So the flow would be:
1. `audit_sink::stop()`
2. `audit_sink::do_toggle(false)`
3. `audit_log_manager::pause()`
4. Pause will drain all shard queues
5. When `pause()` returns `stop()` will be called on the `audit_client`

Reply:
If you have client buffering enabled too on call to `stop()` the client will attempt to flush its buffers as well , even if the thresholds haven't been met: 
https://github.com/redpanda-data/redpanda/blob/dev/src/v/kafka/client/produce_partition.h#L56

Reply:
I think there may be an issue though @BenPope , this just waits 3s for this gate to close, but i think really the intention is that `_client.stop()` will cause the produce_partition to kick off a consume ignoring the buffering options (the line of code i linked above).

But with the changes I've made to the client I believe that once the client hits `send()` it will just fail because the newly added abort source was already triggered. Ideas?

I think the issue are the semantics. With the abort source it indicates "stop what your doing now" but the way the code was originally wrote, `stop()` behaved in a way that would "send whatever your have batched now"